public/app/features/provisioning/Wizard/BootstrapStep.tsx

```diff
@@ -0,0 +1,398 @@
+  onStatusChange: (success: boolean) => void;
+  onRunningChange: (isRunning: boolean) => void;
+  onErrorChange: (error: string | null) => void;
+};
+
+interface OptionState {
+  isDisabled: boolean;
+  disabledReason?: string;
+}
+
+export function BootstrapStep({ onOptionSelect, onStatusChange, onRunningChange, onErrorChange }: Props) {
+  const {
+    register,
+    control,
+    setValue,
+    watch,
+    formState: { errors },
+  } = useFormContext<WizardFormData>();
+
+  const settingsQuery = useGetFrontendSettingsQuery();
+  const currentRepoName = watch('repositoryName');
+  const selectedTarget = watch('repository.sync.target');
+  const [selectedOption, setSelectedOption] = useState<ModeOption | null>(null);
+  const [hasInitialized, setHasInitialized] = useState(false);
+
+  const { value: counts, loading: isLoadingCounts } = useAsync(async () => {
+    onRunningChange(true);
+    try {
+      // Fetch dashboard count
+      const dashboardData = await backendSrv.get('/apis/dashboard.grafana.app/v0alpha1/namespaces/default/search', {
+        query: '*',
+        limit: 0,
+        type: 'dashboard',
+      });
+
+      // Fetch folder count
+      const folderData = await backendSrv.get('/apis/dashboard.grafana.app/v0alpha1/namespaces/default/search', {
+        query: '*',
+        limit: 0,
+        type: 'dashboard',
+      });
+
+      onStatusChange(true);
+      onErrorChange(null);
+
+      return {
+        dashboardCount: dashboardData.totalHits || 0,
+        folderCount: folderData.totalHits || 0,
+      };
+    } catch (error) {
+      console.error('Error fetching counts:', error);
+      onErrorChange(error instanceof Error ? error.message : 'Failed to fetch resource counts');
+      onStatusChange(false);
+      return {
+        dashboardCount: 0,
+        folderCount: 0,
+      };
+    } finally {
+      onRunningChange(false);
+    }
+  }, []);
+
+  // Get repository files count
+  const { data: filesData, isLoading: isLoadingFiles } = useGetRepositoryFilesQuery(
+    { name: currentRepoName || '' },
+    { skip: !currentRepoName }
+  );
+
+  const dashboardCount = counts?.dashboardCount || 0;
+  const folderCount = counts?.folderCount || 0;
+  const fileCount = filesData?.items?.length || 0;
+  const hasLoadedCounts = counts !== undefined;
+
+  // Check for other repositories excluding the current one
+  const [otherInstanceConnected, otherFolderConnected] = useMemo(() => {
+    if (!settingsQuery.data) {
+      return [false, false];
+    }
+
+    const repositories = settingsQuery.data.items || [];
+    // Filter out the current repository if it exists
+    const otherRepos = repositories.filter((repo) => {
+      return repo.name !== currentRepoName;
+    });
+
+    return checkSyncSettings({ ...settingsQuery.data, items: otherRepos });
+  }, [settingsQuery.data, currentRepoName]);
+
+  // Get available options and disabled state logic
+  const getOptionState = useCallback(
```

Comment:
This probably needs to be a function outside this component or even a separate component.

Reply:
so much fight with this one. Having a hard time to have the options enabled and the first available option selected and zero glitches.